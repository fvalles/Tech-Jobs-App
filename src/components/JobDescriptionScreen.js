/* eslint-disable no-undef */
/* eslint-disable no-alert */
import React from 'react';
import { ScrollView } from 'react-native';
import PropTypes from 'prop-types';
import Markdown from 'react-native-markdown-display';
import { StyledText } from './StyledText';
import { StyledTouchableOpacity } from './StyledTouchableOpacity';
import { StyledView } from './StyledView';
import { storeData } from './AsyncStorageHelper';

export default function JobDescriptionScreen({ route }) {
  const { companyName, jobDesc, jobId, jobTitle } = route.params;
  return (
    <ScrollView>
      <StyledView viewType="jobDesc">
        <Markdown>{jobDesc}</Markdown>
      </StyledView>
      <StyledView viewType="flexSaveBtn">
        <StyledTouchableOpacity
          touchableType="descriptionBtn"
          onPress={() => {
            storeData({ id: jobId, companyName, jobDesc, jobTitle });
            alert('Job Saved to Favorites!');
          }}
        >
          <StyledText textType="descBtn">Save Job</StyledText>
        </StyledTouchableOpacity>
      </StyledView>
    </ScrollView>
  );
}

JobDescriptionScreen.propTypes = {
  route: PropTypes.shape({
    params: PropTypes.shape({
      companyName: PropTypes.string.isRequired,
      jobDesc: PropTypes.string.isRequired,
      jobId: PropTypes.string.isRequired,
      jobTitle: PropTypes.string.isRequired,
    }),
  }).isRequired,
};
