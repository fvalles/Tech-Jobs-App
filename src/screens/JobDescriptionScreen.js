/* eslint-disable no-undef */
/* eslint-disable no-alert */
import React from 'react';
import { ScrollView } from 'react-native';
import PropTypes from 'prop-types';
import Markdown from 'react-native-markdown-display';
import { StyledText } from '../components/StyledText';
import { StyledTouchableOpacity } from '../components/StyledTouchableOpacity';
import { StyledView } from '../components/StyledView';
import { storeData } from '../utils/asyncStorageHelper';

export default function JobDescriptionScreen({ route }) {
  const { companyName, isFavJob, jobDesc, jobId, jobTitle } = route.params;
  return (
    <ScrollView>
      <StyledView viewType="jobDesc">
        <Markdown>{jobDesc}</Markdown>
      </StyledView>
      {!isFavJob && (
        <StyledView viewType="flexSaveBtn">
          <StyledTouchableOpacity
            touchableType="descriptionBtn"
            onPress={() => {
              storeData({ id: jobId, companyName, jobDesc, jobTitle });
              alert('Job Saved to Favorites!');
            }}
          >
            <StyledText textType="descBtn">Save Job</StyledText>
          </StyledTouchableOpacity>
        </StyledView>
      )}
    </ScrollView>
  );
}

JobDescriptionScreen.propTypes = {
  route: PropTypes.shape({
    params: PropTypes.shape({
      companyName: PropTypes.string.isRequired,
      isFavJob: PropTypes.bool,
      jobDesc: PropTypes.string.isRequired,
      jobId: PropTypes.string.isRequired,
      jobTitle: PropTypes.string.isRequired,
    }),
  }).isRequired,
};
