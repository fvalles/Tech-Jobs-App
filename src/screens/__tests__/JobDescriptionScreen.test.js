import React from 'react';
import renderer from 'react-test-renderer';

import JobDescriptionScreen from '../JobDescriptionScreen';

describe('<JobDescriptionScreen />', () => {
  const route = {
    params: {
      jobDesc: 'Mock Job Description',
      companyName: 'Mock Company Name',
      jobId: 'Mock Job ID',
      jobTitle: 'Mock Job Title',
    },
  };
  it('has 1 child', () => {
    const tree = renderer.create(<JobDescriptionScreen route={route} />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<JobDescriptionScreen route={route} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
