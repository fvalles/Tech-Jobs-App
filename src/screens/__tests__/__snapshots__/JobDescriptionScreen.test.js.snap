// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobDescriptionScreen /> renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
            "paddingBottom": 5,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 5,
          },
        ]
      }
      viewType="jobDesc"
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginTop": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={Object {}}
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {},
                ]
              }
            >
              Mock Job Description
            </Text>
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "center",
          },
        ]
      }
      viewType="flexSaveBtn"
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#32E0C4",
            "borderRadius": 35,
            "marginBottom": 5,
            "marginTop": 5,
            "opacity": 1,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
            "width": 140,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
          textType="descBtn"
        >
          Save Job
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
